### PR TITLE
Improve/convert to classes

### PR DESCRIPTION
**Pull request by averagengineer**
Split main.cpp in to classes for better code re-usability, maintainability and cleanliness

### CODE REVIEW

File: Command.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    // Process a command
+    int process();
+
+    // Get the output from the command
+    string getOutput();
+
+};
+
+
```
Comment (embluk) [inline Command.h:39]: Can this extra line be removed?

File: Agent.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    string getBakupRequestURL();
+
+    // Generate a bakup job confirmation url
+    string getBakupJobConfirmationURL();
+
+    // Return the working directory
+    string getWorkingDirectory();
+};
+
+
```
Comment (embluk) [inline Agent.h:55]: Can this extra line be removed?

File: Request.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    int getBakupJob();
+
+    // Get the response
+    string getResponse();
+
+    // Get the vector response
+    vector<string> getVectoredResponse();
+};
+
+
```
Comment (embluk) [inline Request.h:49]: Could this extra line be removed?

File: Response.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+#include "cpr/include/cpr/cprtypes.h"
+#include <vector>
+#include "include/rapidjson/include/rapidjson/document.h"
+#include "include/rapidjson/include/rapidjson/writer.h"
+#include "include/rapidjson/include/rapidjson/stringbuffer.h"
+
+using namespace std;
+using namespace rapidjson;
+
+
```
Comment (embluk) [inline Response.h:16]: Extra line here could be removed.

File: main.cpp [hunk 5]
```diff
@@ -229,103 +56,83 @@ int main(int argc, char* argv[])
...
+            if (commandStatusCode != EXIT_SUCCESS)
+            {
+                break;
+            }
         }
-    }
 
-    // End the JSON string
-    writer.EndArray();
+        // End the JSON string
```
Comment (embluk) [inline main.cpp:120]: Accidental left over spacing here?

File: main.cpp [hunk 6]
```diff
@@ -229,103 +56,83 @@ int main(int argc, char* argv[])
...
+        string jobStatusString = s.GetString();
+        debug.Print(jobStatusString);
+        const string jobConfirmationUrl = agent.getBakupJobConfirmationURL();
+        string postJobResponse;
+        Response response(jobConfirmationUrl, agent.getAuthToken());
+        int jobConfStatus = response.postJobConfirmation(jobStatusString, postJobResponse);
 
-    debug.Print(to_string(jobConfStatus));
-    debug.Print(jobResponse);
+        debug.Print(to_string(jobConfStatus));
```
Comment (embluk) [inline main.cpp:130]: Accidental left over spacing here?

**Review by embluk — CHANGES_REQUESTED** [review]
All very good! A move in the right direction with more modularity.

Just some code spacing that can be removed to keep the code consistent :)

File: main.cpp (hunk 5 above)
Comment (averagengineer) [reply main.cpp:120]: Do you mean the indent or the empty lines above and below?

Comment (embluk) [reply main.cpp:120]: The indent.

**Review by embluk — APPROVED** [review]
All solid!

**Pull request merged**